examples/mig/full/variables.tf

```diff
@@ -0,0 +1,248 @@
+}
+
+variable "can_ip_forward" {
+  description = "Enable IP forwarding, for NAT instances for example"
+  default     = "false"
+}
+
+variable "tags" {
+  type        = "list"
+  description = "Network tags, provided as a list"
+  default     = []
+}
+
+variable "labels" {
+  type        = "map"
+  description = "Labels, provided as a map"
+  default     = {}
+}
+
+/* disk */
+variable "source_image" {
+  description = "Source disk image. If neither source_image nor source_image_family is specified, defaults to the latest public CentOS image."
+  default     = ""
+}
+
+variable "source_image_family" {
+  description = "Source image family. If neither source_image nor source_image_family is specified, defaults to the latest public CentOS image."
+  default     = ""
+}
+
+variable "source_image_project" {
+  description = "Project where the source image comes from"
+  default     = ""
+}
+
+variable "disk_size_gb" {
+  description = "Disk size in GB"
+  default     = "100"
+}
+
+variable "disk_type" {
+  description = "Disk type, can be either pd-ssd, local-ssd, or pd-standard"
+  default     = "pd-standard"
+}
+
+variable "auto_delete" {
+  description = "Whether or not the disk should be auto-deleted"
+  default     = "true"
+}
+
+variable "additional_disks" {
+  description = "List of maps of additional disks. See https://www.terraform.io/docs/providers/google/r/compute_instance_template.html#disk_name"
+  type        = "list"
+  default     = []
+}
+
+/* metadata */
+variable "startup_script" {
+  description = "User startup script to run when instances spin up"
+  default     = ""
+}
+
+variable "metadata" {
+  type        = "map"
+  description = "Metadata, provided as a map"
+  default     = {}
+}
+
+/* service account */
+variable "service_account" {
+  type        = "map"
+  description = "Service account email address and scopes"
+}
+
+##########################
+# Mananged Instance Group
+##########################
+
+variable "target_size" {
+  description = "The target number of running instances for this managed or unmanaged instance group. This value should always be explicitly set unless this resource is attached to an autoscaler, in which case it should never be set."
+  default     = 1
+}
+
+variable "target_pools" {
+  description = "The target load balancing pools to assign this group to."
+  type        = "list"
+  default     = []
+}
+
+variable "distribution_policy_zones" {
+  description = "The distribution policy, i.e. which zone(s) should instances be create in. Default is all zones in given region."
+  type        = "list"
+  default     = []
+}
+
+variable "update_policy" {
+  description = "The rolling update policy. https://www.terraform.io/docs/providers/google/r/compute_region_instance_group_manager.html#rolling_update_policy"
+  type        = "list"
+  default     = []
+}
+
+/* health checks */
+
+variable "http_healthcheck_enable" {
+  description = "Enable HTTP healthcheck"
+  default     = false
```

Comment:
Boolean values of variables must be quoted to ensure interoperability with Terraform v0.12.